quinn-proto/src/connection/mtud.rs

```diff
@@ -50,8 +47,7 @@ impl MtuDiscovery {
         Self {
             current_mtu,
             state,
-            black_hole_counter: 0,
-            largest_acked_mtu_sized_packet: None,
+            black_hole_detector: BlackHoleDetector::new(),
```

Comment:
Might not make sense now, but for the future: this would be easier to review as two commits, one that stuffs the current logic into the a `BlackHoleDetector` type, and one that replaces the `BlackHoleDetector` implementation.

Reply:
Thanks, I'll keep it in mind